### ISSUE TITLE
how to load custom dataset for this example? 

### ISSUE THREAD

**Issue by smith-co** [anukash/Keras_siamese_contrastive#2]
@anukash thanks for giving feedback [here](https://github.com/anukash/Keras_siamese_contrastive/issues/1).

However, I would be interested to apply `contrastive learning` approach and not `triplet loss` technique. Thats why I was trying to use code from your repo.

I would follow the `pre_process(file_path)` example from https://github.com/anukash/Face_recognition_siamese_network.

But would the rest of the code i.e. model would work as is?

**Comment by smith-co** [comment]
Alternately I suppose if I change the loss function in the [Face_recognition_siamese_network](https://github.com/anukash/Face_recognition_siamese_network) from triplet loss to `contrastive_loss` I suppose that would be contrastive learning - right?

**Comment by anukash** [comment]
Yes I guess, Change will come at loss function part and L1_dist layer,
Rest of code might not work you have to change the codes accordingly one thing you can try is to take the data preparation part from one repo and model making and calling path from other, I hope that work.

Kindly let me know the results of your experiments.

**Comment by smith-co** [comment]
If I understand it correctly:

`data preparation part`: https://github.com/anukash/Face_recognition_siamese_network/blob/main/train.py#L25
`model`:https://github.com/anukash/Keras_siamese_contrastive/blob/d360e8db3b2032be7f7d84f9b4f95d3eac86b4fb/siamese_train.py#L177

Alternately I am thinking to change the `loss function part` and `L1_dist layer` in https://github.com/anukash/Face_recognition_siamese_network/ here for contrastive learning.

@anukash which one do you think would be easier to implement?

I am also super curious to know how the triplet loss vs contrastive learning works. I would definitely let you know if I could get it working.

**Comment by anukash** [comment]
> Alternately I suppose if I change the loss function in the [Face_recognition_siamese_network](https://github.com/anukash/Face_recognition_siamese_network) from triplet loss to `contrastive_loss` I suppose that would be contrastive learning - right?

According to what I have read Triplet loss is used when we have three images and contrastive loss work on pair of images both called as distance based loss function
One advantage of the triplet loss is that it tries to be less “greedy” than the contrastive loss (which considers pairwise examples). This is because the triplet loss takes an anchor example and tries to bring positive examples closer while also pushing away negative example

**Comment by anukash** [comment]
It will work, Instead of seeing as three separate image vector problem see it as two vector which are distance between positive with anchor and negative with anchor.
I think this will full code will work https://github.com/anukash/Face_recognition_siamese_network, Only change here is loss, and as you are going to use a custom loss function(contrastive loss) after defining it in training, Define the custom loss fucntion in test as well it will work. check this https://github.com/anukash/Keras_siamese_contrastive  for loading custom layer and loss, its done with minimal changes.

**Comment by smith-co** [comment]
yeah - that's what I read as well. But would be super interesting to try with both and observe the result.

**Comment by smith-co** [comment]
> It will work, Instead of seeing as three separate image vector problem see it as two vector which are distance between positive with anchor and negative with anchor.
> I think this will full code will work https://github.com/anukash/Face_recognition_siamese_network, Only change here is loss, and as you are going to use a custom loss function(contrastive loss) after defining it in training, Define the custom loss fucntion in test as well it will work. check this https://github.com/anukash/Keras_siamese_contrastive for loading custom layer and loss, its done with minimal changes.

Looks like there are lots of issues with tensor etc coming up 😢
